data/Builder.go

```diff
@@ -0,0 +1,110 @@
+package data
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+)
+
+//Datum represent one data point
+type Datum map[string]interface{}
+
+//Dataset represents an array of data points
+type Dataset []Datum
+
+//GetSelector returns the selector fields for a given platform datatype, or nil if there is no match
+func GetSelector(data interface{}) interface{} {
+
+	switch data.(type) {
+	case *Basal:
+		return data.(*Basal).Selector()
+	case *DeviceEvent:
+		return data.(*DeviceEvent).Selector()
+	default:
+		return nil
+	}
+}
+
+//Builder interface that the TypeBuilder implements
+type Builder interface {
+	BuildFromRaw(raw []byte) (interface{}, *Error)
+	BuildFromData(data map[string]interface{}) (interface{}, *Error)
```

Comment:
Use `Datum` as type rather than `map[string]interface{}`.  What does this return, BTW, I see it returns an interface{} but should we have a real type for that?